### ISSUE TITLE
GPIO40 Unselectable

### ISSUE THREAD

**Issue by keabarnes** [jgarff/rpi_ws281x#209]
Set LED_PIN = 40 in strandtest.py, running on Compute Module 3, GPIO40 is PWM0.

Looking at the code, it would seem the number 40 needs to be added to gpionums_40p array in ws2811.c.

Error:
```Gpio 40 is illegal for LED channel 0

Traceback (most recent call last):

  File "strandtest.py", line 87, in <module>

    strip.begin()

  File "build/bdist.linux-armv7l/egg/neopixel.py", line 115, in begin

RuntimeError: ws2811_init failed with code -11 (Selected GPIO not possible)

**Comment by chuchaki** [comment]
Hey,

I have exactly the same error, only that it already appears using the default pwm (gpio18). Im using an old rpi B rev 2 (2011.12). Are there any known issues that could cause that problem?

**Comment by chuchaki** [comment]
Follow up:
The error is thrown by the method '_static int check_hwver_and_gpionum(ws2811_t *ws2811)' int the file 'https://github.com/jgarff/rpi_ws281x/blob/master/ws2811.c]'  at line 665.

It separates by the corresponding hardware revision code (/proc/cpuinfo). This list ( http://www.raspberrypi-spy.co.uk/2012/09/checking-your-raspberry-pi-board-version/ ) states that model A has the revision number 0x008. The code in line 687 indicates that the model A is selected by the else if branch. 0x00e - 0x00f indicates, that only model B rev 2 is possible here. Is this an error, or is model A not supported.

**Comment by tvoverbeek** [comment]
@chuchaki  See my PR #222. Model A should be recognised now.

**Comment by tvoverbeek** [comment]
@keabarnes Also included CM1 and CM3 support in PR #222.
Cannot test it myself, but GPIO40 should work with this PR

**Comment by jgarff** [comment]
Merged into mainline.  Please retry with the latest and reopen if necessary.  Thanks.
